### PR TITLE
[IMP] hr_expense_invoice: onchange invoice

### PR DESCRIPTION
**Pull request by chienandalu**
Get expense amount from invoice amount. Otherwise it will do a mismatch when trying to post the account move.

cc @Tecnativa TT25442

### CODE REVIEW

**Review by pedrobaeza — APPROVED** [review]

**Comment by chienandalu** [comment]
@pedrobaeza I added a little change for defaulting to the invoice analytic account as well

File: hr_expense_invoice/models/hr_expense.py [hunk 1]
```diff
@@ -38,3 +37,14 @@ def _get_account_move_line_values(self):
...
+    @api.onchange("invoice_id")
+    def _onchange_invoice_id(self):
+        """Get expense amount from invoice amount. Otherwise it will do a
+           mismatch when trying to post the account move. We also default
+           to the invoice analytic account although it can be changed if
+           needed"""
+        if self.invoice_id:
+            self.quantity = 1
+            self.unit_amount = self.invoice_id.amount_total
+            self.analytic_account_id = self.invoice_id.analytic_account_id
```
Comment (pedrobaeza) [inline hr_expense_invoice/models/hr_expense.py]: AFAIK, there's no analytic account at invoice level, but line level.

Comment (chienandalu) [reply hr_expense_invoice/models/hr_expense.py]: True. I took it for granted... changed

Comment (pedrobaeza) [reply hr_expense_invoice/models/hr_expense.py]: Anyway, it should be the same, as this module overrides account.move.line creation for linking the already created ones in invoice, so the analytic account will be used. Am I correct?

Comment (chienandalu) [reply hr_expense_invoice/models/hr_expense.py]: But we'd have the problem with multiple lines. Wich one has the correct analytic account?

Comment (pedrobaeza) [reply hr_expense_invoice/models/hr_expense.py]: It doesn't matter, as the journal items will be generated from the invoice, not from the expense. Take into account that the analytic account goes to the expense account, not to the payable.

**Review by victoralmau — APPROVED** [review]

**Review by rafaelbn — CHANGES_REQUESTED** [review]
Although the analytical account is informed. And also the accounting account gives an error.

**Review by rafaelbn — APPROVED** [review]
Functionall review :+1: 

Previous commment was with demo data.

RTM

**Comment by rafaelbn** [comment]
@kittiu can you review to get this merge?

**Review by kittiu — APPROVED** [review]
LGTM

**Review by marcelsavegnago — COMMENTED** [review]
LGTM

**Pull request merged**